src/Uno.UWP/Networking/Connectivity/NetworkInformation.Android.cs

```diff
@@ -0,0 +1,147 @@
+		}
+
+		private static void StopNetworkStatusChanged()
+		{
+			if (_connectivityChangeBroadcastReceiver == null)
+			{
+				return;
+			}
+
+			Application.Context.UnregisterReceiver(
+				_connectivityChangeBroadcastReceiver);
+			_connectivityChangeBroadcastReceiver?.Dispose();
+			_connectivityChangeBroadcastReceiver = null;
+		}
+
+		public static IReadOnlyList<HostName> GetHostNames()
+		{
+
+			Android.OS.StrictMode.ThreadPolicy prevPolicy = null;
+
+			if (Android.OS.Build.VERSION.SdkInt > Android.OS.BuildVersionCodes.Gingerbread)
+			{
+				// required to access interfaceAddress.Address.CanonicalHostName (below), without creating new task, outside UI thread
+				prevPolicy = Android.OS.StrictMode.GetThreadPolicy();
+				var policy = new Android.OS.StrictMode.ThreadPolicy.Builder().PermitAll().Build();
+				Android.OS.StrictMode.SetThreadPolicy(policy);
+			}
+
+
+			var uwpList = new List<HostName>();
+
+			var netInterfacesEnum = Java.Net.NetworkInterface.NetworkInterfaces;
+			while (netInterfacesEnum.HasMoreElements)
+			{
+
+				var netInterface = netInterfacesEnum.NextElement() as Java.Net.NetworkInterface;
+				if (netInterface.InterfaceAddresses != null)
+				{
+					string androDisplayName = netInterface.DisplayName;
+
+					// another name, netInterface.Name, would be ignored (seems like == androDisplayName)
+					foreach (var interfaceAddress in netInterface.InterfaceAddresses)
+					{
+						int androPrefixLength = interfaceAddress.NetworkPrefixLength;
+						if (interfaceAddress.Address != null)
+						{
+							string androCanonical = interfaceAddress.Address.CanonicalHostName;
+							// seems like == androCanonical
+							string androHostName = interfaceAddress.Address.HostName;
+							bool androIPv46 = (interfaceAddress.Address.GetAddress().Count() == 4);
+
+							// we have all required data from Android, and we can use them
+							HostName newHost = new HostName();
+							IPInformation newInfo = new IPInformation();
+							newInfo.PrefixLength = (byte)androPrefixLength;
+
+							newHost.IPInformation = newInfo;
+
+							// only these two types; UWP has also 'DomainName' and 'Bluetooth'
+							newHost.Type = (androIPv46) ? HostNameType.Ipv4 : HostNameType.Ipv6;
+							
+							newHost.CanonicalName = androCanonical;
+							newHost.RawName = androHostName;
+							newHost.DisplayName = androDisplayName;
+							// assuming there would be no duplicates
+							uwpList.Add(newHost);
+						}
+					}
+				}
+			}
+
+			if (Android.OS.Build.VERSION.SdkInt > Android.OS.BuildVersionCodes.Gingerbread)
+			{
+				// returning old policy
+				Android.OS.StrictMode.SetThreadPolicy(prevPolicy);
+			}
+
+			return uwpList;
+		}
+
+		/// <summary>
+		/// This raises the NetworkStatusChanged event with a delay.
+		/// Delay is necessary as network information is not updated before
+		/// the connectivity broadcast on Android is received.
+		/// </summary>
+		internal static async void OnDelayedNetworkStatusChanged()

```

Comment:
Don't use async void

Reply:
I need to delay to make sure the network information is actually up to date when the event is raised. I can use `async Task` but it still must be fire and forget, as it will originate from the broadcast receiver.

Reply:
Fixed :-)